lightning/src/onion_message/packet.rs

```diff
@@ -135,6 +135,9 @@ impl<T: OnionMessageContents> OnionMessageContents for ParsedOnionMessageContent
 			&ParsedOnionMessageContents::Custom(ref msg) => msg.tlv_type(),
 		}
 	}
+	fn msg_type() -> &'static str {
```

Comment:
Hmm... this was requested to avoid having users pass a string to `handle_onion_message_response` in the follow-up. See https://github.com/lightningdevkit/rust-lightning/pull/2907#discussion_r1569541966. We really only care about the class of message we are responding to. In `handle_onion_message_response`, we have the response (not the original request), so calling this on the response message passed in wouldn't give a correct logging. That is, for example, we currently log "when responding to Offers onion message" rather than "when responding to InvoiceRequest onion message". And by making this take `&self` it would log "when responding to Invoice onion message", which is not correct.

Reply:
This assumes the `OffersMessage` implementation changed. Though for custom messages, I can see how a handler may implement more than one class of messages. So I'm not sure how to best handle this. Either in the async case the user passes in the request message's string or we rephrase the logging maybe?

Reply:
I guess we could hold the request string in `Responder` and move it to the `ResponseInstructions`.That way the user doesn't need to pass it. @shaavan Let's handle this in the follow-up PR.

Reply:
As mentioned above, the log is stating which message the response is in response to, not the response itself.

Reply:
The rest of the message says "with path_id {:02x?}", which refers to the message being respond to. Anyhow, you're right that we should change it to `&self`. Doing so would allow implementations to return the specific message type.

Reply:
I think the suggestion sounds reasonable!
Reverted the `msg_type` function to take `self` as a reference in [32a8615](https://github.com/lightningdevkit/rust-lightning/commit/32a8615de62bc63439b529b827e016d5bff48aa0)
Thanks for the pointers!